langlib/langlib-test/src/test/resources/test-src/functionlib_test.bal

```diff
@@ -228,28 +228,28 @@ function testCallFunctionWithInvalidArguments() {
     a1 = trap function:call(value8, "");
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[int...]', found '[string]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: argument(s) of type(s) 'int...', cannot be passed to function expecting parameter(s) of type(s) 'string'", (<error> a1).detail()["message"]);
 
     a1 = trap function:call(value8, "", "");
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[int...]', found '[string,string]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: argument(s) of type(s) 'int...', cannot be passed to function expecting parameter(s) of type(s) 'string,string'", (<error> a1).detail()["message"]);
 
     a1 = trap function:call(value1, 10);
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[int,int,int]', found '[int]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: argument(s) of type(s) 'int,int,int', cannot be passed to function expecting parameter(s) of type(s) 'int'", (<error> a1).detail()["message"]);
 
     NewPerson person = {firstName: "chiran", secondName: "sachintha"};
     a1 = trap function:call(test10, person);
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[string,NewPerson]', found '[NewPerson]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: argument(s) of type(s) 'string,NewPerson', cannot be passed to function expecting parameter(s) of type(s) 'NewPerson'", (<error> a1).detail()["message"]);
 
     a1 = trap function:call(test11, {});
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[NewStudent]', found '[map<(any|error)>]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: argument(s) of type(s) 'NewStudent', cannot be passed to function expecting parameter(s) of type(s) 'map<(any|error)>'", (<error> a1).detail()["message"]);
```

Comment:
Don't we have tests with rest params?

Reply:
Added negative test case with rest parameter.